app/services/error_queue_handler.rb

```diff
@@ -0,0 +1,108 @@
+class ErrorQueueHandler
+  def message_count
+    count = 0
+    with_error_queue do |error_queue|
+      count = error_queue.message_count
+    end
+    count
+  end
+
+  def messages(routing_key: nil, limit: nil)
+    msgs = []
+    each_error_queue_message do |msg|
+      msgs << serialize_message(msg) unless routing_key &&
+        msg.first[:routing_key] != routing_key
+      break if limit && msgs.length == limit
+    end
+    msgs
+  end
+
+  def requeue_message(id)
+    message = nil
+    msgs = []
+    each_error_queue_message do |msg, channel, delivery_tags|
+      msgs << serialize_message(msg)
+      if msgs.last[:id] == id
+        republish_message(channel, msgs.last)
+        channel.ack(delivery_tags.pop)
+        message = msgs.last
+        break
+      end
+    end
+    message
+  end
+
+  def requeue_all
+    msgs = []
+    each_error_queue_message do |msg, channel, delivery_tags|
+      msgs << serialize_message(msg)
+      republish_message(channel, msgs.last)
+      channel.ack(delivery_tags.pop)
+    end
+    msgs
+  end
+
+  def requeue_messages(routing_key:, limit: nil)
+    msgs = []
+    each_error_queue_message do |msg, channel, delivery_tags|
+      if routing_key && msg.first[:routing_key] == routing_key
```

Comment:
either your routing_key keyword argument needs a default (e.g. requeue_messages(routing_key: nil,...) or you can remove the extra test for routing_key because the method will always throw an ArgumentError if routing_key is not defined in the arguments.